### PR TITLE
conky: init latest

### PR DESCRIPTION
**Pull request by lovek323**
###### Motivation for this change

I am not sure why conky haven't released a new version, but this issue was fixed in April: https://github.com/brndnmtthws/conky/issues/471 (with this commit: https://github.com/brndnmtthws/conky/pull/479/commits/8790281fa56eb0ec5e2ecf992e66cee94208d878)

I use docker and conky and without this fix it's broken.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

I'm not sure where to place things like this or if there's a better way to create a new derivation that only changes the version and the hash. Any advice appreciated.

### CODE REVIEW

**Comment by erikarvstedt** [comment]
There's no need to duplicate the whole package definition. `conky/latest.nix` can be written like so:
```
{ conky, fetchFromGitHub }:
                                                                                                                                                          
conky.overrideAttrs (_: rec {
  name = "conky-${version}";
  version = "f8ff46c";
                                                                                                                                                        
  src = fetchFromGitHub {
    owner = "brndnmtthws";
    repo = "conky";
    rev = "f8ff46c2dca4d639c9287790c35999bbaae56010";
    sha256 = "0k926axqm1f7pm676lz12ibmcf6c2haiz90sxqsgcp0lb1p58jz1";
  };
});
```

**Comment by worldofpeace** [comment]
Also the version of this package should be in the format `unstable-YYYY-MM-DD` with that date being from the commit.
> ##### https://nixos.org/nixpkgs/manual/#sec-package-naming

Also are we sure we can just need to override the default conky?
Lots of things have changed in the [118 commits](https://github.com/brndnmtthws/conky/compare/v1.10.8...master)  since the last release.

**Comment by worldofpeace** [comment]
> > Also are we sure we can just need to override the default conky?
> > Lots of things have changed in the 118 commits since the last release.
> 
> @worldofpeace, Are you suggesting that I just replace the default with the latest `master`? I've been using this version of conky for a few hours now and it is working fine for me, so I'm happy to do that.


Ahh let me rephrase that. 

I was suggesting that they could have added more build options, for example, that we need to make configurable. Also it appears that [darwin support](https://github.com/brndnmtthws/conky/commit/b20d83b747fd074963c605f2177c0b3205b5e8b7) is in the works.
So it's not a straightforward update where we can just override the src.

And that https://github.com/brndnmtthws/conky/compare/v1.10.8...master could be helpful to you, so you can decide what's needed.

**Comment by Mic92** [comment]
Can you try to pull the patches in question into our conky instead of adding a new package?

**Comment by lovek323** [comment]
> Can you try to pull the patches in question into our conky instead of adding a new package?

@Mic92: Yep, I could pull just that fix in. That might be easiest.

**Comment by Mic92** [comment]
You can ping the conky maintainer in a bit instead of adding an unstable version of this package.

**Comment by lovek323** [comment]
Closing awaiting a new conky release.
